streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -72,8 +82,23 @@ public StreamThreadStateStoreProvider(final StreamThread streamThread) {
         } else {
             throw new InvalidStateStoreException("Cannot get state store " + storeName + " because the stream thread is " +
                                                      state + ", not RUNNING" +
-                                                     (includeStaleStores ? " or REBALANCING" : ""));
+                                                     (storeQueryParams.staleStoresEnabled() ? " or REBALANCING" : ""));
         }
     }
 
+    private TaskId createKeyTaskId(final String storeName, final Integer partition) {
+        if (partition == null) {
+            return null;
+        }
+        final List<String> sourceTopics = internalTopologyBuilder.stateStoreNameToSourceTopics().get(storeName);
```

Comment:
If we only need the source topics names and topic groups, we should not pass in the full `InternalTopologyBuilder` into this class, but only the two pieces we need.

Reply:
`internalTopologyBuilder.stateStoreNameToSourceTopics()` may return different values over time if the source nodes are regex pattern which means during rebalances we may update the stateStoreNameToSourceTopics mapping.

Reply:
However the below `topicGroups()` can evolve as well if we have regex source nodes, also for the `stateStoreNameToSourceTopics` it is only used in `StreamsMetadataState` to verify / build the metadata. I think we can actually have this logic inside `InternalTopologyBuilder` and just expose a

```
List<String> getSourceTopicsForStore(String storeName)
```

 that both here and `StreamsMetadataState` can use. Then we do not need to overkill `stateStoreNameToSourceTopics` anymore.

This is just a code style suggestion.

Reply:
Thanks @guozhangwang -- as this is internal, we can still improve on it in a follow up.

Reply:
Thanks @guozhangwang and @mjsax on suggestions here. Now, since these can change during rebalancing, I am assuming just passing source topics names and topic groups will not suffice and we will have to pass complete internalTopologyBuilder. Is that right? I have added `List<String> getSourceTopicsForStore(String storeName)` in InternalTopologyBuilder class though.